lib/exhal/navigation.ex

```diff
@@ -42,7 +42,10 @@ defmodule ExHal.Navigation do
 
     case ExHal.get_links_lazy(a_doc, name, fn -> :missing end) do
       :missing -> missing_link_handler.(name)
-      links -> Enum.map(links, &_follow_link(a_doc.client, &1, tmpl_vars, opts))
+      links ->
+        links
+        |> Task.async_stream(&_follow_link(a_doc.client, &1, tmpl_vars, opts))
+        |> Enum.map(fn {:ok, followed_link} -> followed_link end)
```

Comment:
Codebeat is complaining about the length of this function. What if we changed its signature to `follow_links(a_doc, name, missing_link_handler, tmpl_vars: tmpl_vars)` and moved the Map unpacking to another function head?

Reply:
It seems like we still need to pass `opts` down to `@client_module` for headers and `Poison` options. I could simply extract the async and map into a private function to satisfy Codebeat.

Reply:
I kinda like having this this function coordinate the work. Maybe we pull out the opt unpacking in to a private function? eg, `{tmpl_vars, opts} = interpret_follow_links_opts(opts)`

Reply:
Pulled out into `tmpl_vars_and_opts_map/1`